doc/sonic-installer/SONiC_Image_Installation_Verification.md

```diff
@@ -0,0 +1,379 @@
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --total-mem-threshold INTEGER   If system total memory is lower than
+                                  threshold, setup SWAP memory  [default:
+                                  (2048 MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --available-mem-threshold INTEGER
+                                  If system available memory is lower than
+                                  threhold, setup SWAP memory  [default: (1200
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --help                          Show this message and exit.
+
+```
+
+# 4 Design and Flow Diagrams
+
+## 4.1 Background
+
+### 4.1.1 Design Attempts
+
+**Trial One: Simply use ASIC as comparison**
+
+In SONiC, the current ASIC info can be accessed from `/etc/sonic/sonic_version.yml`. With that, we can compare the `asic_type` with the to-be-installed image's ASIC and check if they match.
+```
+admin@str-s6000-acs-13:~$ cat /etc/sonic/sonic_version.yml
+...
+asic_type: broadcom                         <======
+```
+Need to mention that, it may prompt false negative message. Assume we have an broadcom switch which has already installed mellanox built image as no ASIC check before. Its `asic_type` will be translated to mellanox. Then it will wrongly report asic doesn't match if we try to install the correct broadcom built image.
+
+In ONiE, that info is not accessible. So we cannot do the ASIC check in ONiE.
+```
+ONIE:/ # cat etc/machine.conf
+onie_version=3.20.1.5
+onie_vendor_id=674
+onie_platform=x86_64-dell_s6000_s1220-r0
+onie_machine=dell_s6000_s1220
+onie_machine_rev=0
+onie_arch=x86_64
+onie_config_version=1
+onie_build_date="2016-02-09T17:14-0800"
+onie_partition_type=gpt
+onie_kernel_version=3.15.10
+onie_firmware=auto
+```
+So we turn to Trial two.
+
+**Trial Two: Build up a mapping for platform string->ASIC**
+
+In both ONiE and SONiC, the platform string is accessible.
+```
+ONIE:/ # cat etc/machine.conf
+...
+onie_platform=x86_64-dell_s6000_s1220-r0    <======
+```
+```
+admin@str-s6000-acs-13:~$ cat /host/machine.conf
+...
+onie_platform=x86_64-dell_s6000_s1220-r0    <======
+```
+The mapping will be carried in our built image and look like this:
+```
+x86_64-dell_s6000_s1220-r0=broadcom
+x86_64-dell_s6100_c2538-r0=broadcom
+x86_64-mlnx_msn2410-r0=mellanox
+x86_64-mlnx_msn2700-r0=mellanox
+...
+```
+Pros:
+ - The mapping fits both ONiE and SONiC installer. During installation, we can derive current platform's expected ASIC type based on the platform string, then compare that with the to-be-installed image's ASIC.
+
+Cons:
+ - It is a static mapping, which means that the mapping need to be kept in all repos and all ASIC build.
```

Comment:
>all repos [](http://example.com/codeflow?start=75&length=9)

What do you mean by "all repos"? Normally we mean git repo, and the feature is only relevant on sonic-buildimage repo and sonic-utilities repo.

Reply:
I mean we have public repo and internal repo. The platforms we support are somehow different.